src/Receipt/Receipt.php

```diff
@@ -1,21 +1,20 @@
 <?php
+
 namespace Give\Receipt;
 
-use Give\Receipt\DonationDetailsGroup\DonationDetailsGroup;
-use Give\Receipt\AdditionalDetailsGroup\AdditionalDetailsGroup;
-use Give\Receipt\DonorDetailsGroup\DonorDetailsGroup;
+use ArrayAccess;
+use Iterator;
 
 /**
  * Class Receipt
  *
  * This class represent receipt as object.
- * Receipt can have multiple detail group and detail group can has multiple detail item.
- * You can add your own logic to render receipt because it does not return any UI item.
+ * Receipt can have multiple sections and sections can have multiple line items.
  *
  * @since 2.7.0
  * @package Give\Receipt
  */
-class Receipt {
+abstract class Receipt implements Iterator, ArrayAccess {
```

Comment:
Go ahead and move the `Iterator` and `ArrayAccess` methods into this abstract class. They're not specific to donations.

Reply:
fixed: https://github.com/impress-org/givewp/compare/ccb59a12b626...55c57984a9f5